### PR TITLE
PC check_registercloudguest improvements

### PR DESCRIPTION
**Pull request by pdostal**
* Retry after deregistering the system before checking the present repositories
* Count the number of present repositories from the XML output
   * We cannot check the `/etc/zypp/repos.d/` folder as it does not contain all the repositories (because of zypper services)
   * The `zypper lr` contains additional lines and those are not easy to filter out.

- Related ticket: [poo#155776](https://progress.opensuse.org/issues/155776)
- Verification run: [EC2](https://pdostal-server.suse.cz/tests/5982), [Azure](https://pdostal-server.suse.cz/tests/5983), [GCE](https://pdostal-server.suse.cz/tests/5984)

### CODE REVIEW

**Review by asmorodskyi — APPROVED** [review]
LGTM , two small comments feel free to ignore them : 
1. `zypper lr` does not require sudo can be executed by normal user too
2. I am not sure that we need to give zypper 10 minutes to list local repos :thinking:

**Review by ricardobranco777 — APPROVED** [review]

File: tests/publiccloud/check_registercloudguest.pm [hunk 1]
```diff
@@ -76,7 +76,7 @@ sub run {
             $instance->ssh_assert_script_run(cmd => '! sudo SUSEConnect -d', fail_message => 'SUSEConnect succeeds but it is not supported should fail on BYOS');
         }
     } else {
-        if ($instance->ssh_script_output(cmd => 'sudo zypper lr | wc -l', timeout => 360) < 5) {
+        if ($instance->ssh_script_output(cmd => 'sudo zypper -x lr | grep "<repo" | wc -l', timeout => 360) < 3) {
```
Comment (clanig) [inline tests/publiccloud/check_registercloudguest.pm:79]: TBH, I do not understand why we want to ensure that there are only three lines of repositories displayed. This not being the case does not appear as a valid criterion for a test to fail. It does not appear a breaking criterion for an OS to have a limited number of repos available...

Is it possible to check for anything that is more substantial?

**Comment by pdostal** [comment]
>     2. I am not sure that we need to give zypper 10 minutes to list local repos 🤔

Are you aware that zypper has services and those might need to look up the list of repositories online?
So for `zypper lr` we as far as I know might need to be online. And it may also take some time.

File: tests/publiccloud/check_registercloudguest.pm (hunk 1 above)
Comment (clanig) [reply tests/publiccloud/check_registercloudguest.pm:79]: Sorry for the misunderstanding from my side, indeed  it checks if there are at least 3 repos and not at max.

We can handle the clarification in another ticket.

**Comment by asmorodskyi** [comment]
> > ```
> > 2. I am not sure that we need to give zypper 10 minutes to list local repos 🤔
> > ```
> 
> Are you aware that zypper has services and those might need to look up the list of repositories online? So for `zypper lr` we as far as I know might need to be online. And it may also take some time.

IMHO `zypper lr` is NOT doing any external call and not doing `zypper ref` call in background . this is why it also can be done by regular user . 

BUT like I said it is NOT topic for this PR because we discussing not something what you have changed but something what was there before ...

**Review by asmorodskyi — APPROVED** [review]
:+1:

**Review by clanig — APPROVED** [review]

**Pull request merged**